src/client/auth/mod.rs

```diff
@@ -319,7 +319,7 @@ pub struct Credential {
 }
 
 impl Credential {
-    #[cfg(test)]
+    #[cfg(all(test, not(feature = "sync")))]
```

Comment:
The connection string and URI options tests make heavy use of `ClientOptions::parse`. Rather than adding conditional compilation to the tests themselves when needed, it was simpler to just disable those tests for the `sync` feature, as we [only run the `sync` module tests](https://github.com/mongodb/mongo-rust-driver/blob/b0e59f9fde2e5bc3df3217dec05ffbca553d1b8e/.evergreen/run-async-std-tests.sh#L15) with the `sync` feature anyhow.

This method (and a few others) are only used in the connection string/URI options tests, so I updated the conditional compilation to omit them if the sync feature is enabled.